### PR TITLE
Make the infraction message mention @ModMail

### PR DESCRIPTION
**Pull request by swfarnsworth**
Previously, the footer for non-ban infraction messages would say to DM the ModMail bot. This commit makes "ModMail" a clickable mention of the bot.

image

When this footer was introduced, we decided not to make it a mention, in case the recipient did not have the ModMail bot account in their user cache (which would cause it to render as the account ID, ie "send a message to 683001325440860340"). Chris has determined that because the bot is at the top of the user list, it seems to always be in the cache for any guild member, even without visiting the guild or viewing the user list.

### CODE REVIEW

**Review by ChrisLovering — APPROVED** [review]

File: bot/exts/moderation/infraction/_utils.py [hunk 1]
```diff
@@ -31,12 +31,13 @@
...
 
 APPEAL_SERVER_INVITE = "https://discord.gg/WXrCJxWBnm"
+MODMAIL_ACCOUNT_ID = "683001325440860340"
 
 INFRACTION_TITLE = "Please review our rules"
 INFRACTION_APPEAL_SERVER_FOOTER = f"\nTo appeal this infraction, join our [appeals server]({APPEAL_SERVER_INVITE})."
 INFRACTION_APPEAL_MODMAIL_FOOTER = (
     '\nIf you would like to discuss or appeal this infraction, '
-    'send a message to the ModMail bot.'
+    f'send a message to <@{MODMAIL_ACCOUNT_ID}>.'
```
Comment (vivekashok1221) [inline bot/exts/moderation/infraction/_utils.py:40]: How about having the mention inside a pair of parentheses instead?
```suggestion
    f'send a message to the ModMail bot (<@{MODMAIL_ACCOUNT_ID}>).'
```

**Review by vivekashok1221 — CHANGES_REQUESTED** [review]

File: bot/exts/moderation/infraction/_utils.py (hunk 1 above)
Comment (CaedenPH) [inline bot/exts/moderation/infraction/_utils.py:40]: What is the difference when having `<@{MODMAIL_ACCOUNT_ID}>` opposed to `<@!{MODMAIL_ACCOUNT_ID}>`?

Comment (ChrisLovering) [reply bot/exts/moderation/infraction/_utils.py:40]: Since this is sent in a DM channel, the bot doesn't have a nickname, so the ! will do nothing right?

Comment (swfarnsworth) [reply bot/exts/moderation/infraction/_utils.py:40]: image

This is how `<@!683001325440860340>` rendered for me in my DMs with the Python bot. But <@683001325440860340>` renders the same way.

Comment (wookie184) [reply bot/exts/moderation/infraction/_utils.py:40]: I don't mind either way, thoughts @swfarnsworth?

Comment (vivekashok1221) [reply bot/exts/moderation/infraction/_utils.py:40]: I don't really mind either way too, for the  record.

**Comment by swfarnsworth** [comment]
For whatever this is worth: I just tested this in my test server by warning a dummy account that hasn't viewed the main guild in a long time, and the ModMail bot was not in that account's user cache until I clicked through to the guild.

**Review by vivekashok1221 — APPROVED** [review]
LGTM :star2: .

**Comment by wookie184** [comment]
> For whatever this is worth: I just tested this in my test server by warning a dummy account that hasn't viewed the main guild in a long time, and the ModMail bot was not in that account's user cache until I clicked through to the guild.

Ah interesting, it probably is good that we mention it's the modmail bot then 👍

**Review by wookie184 — APPROVED** [review]
Thanks

**Pull request merged**